test/integration/driver_install_or_update_darwin_test.go

```diff
@@ -0,0 +1,124 @@
+package integration
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/tools/apidiff/ioext"
+
+	"k8s.io/minikube/pkg/version"
+)
+
+func TestHyperkitDriverSkipUpgrade(t *testing.T) {
+	MaybeParallel(t)
+	tests := []struct {
+		name            string
+		path            string
+		expectedVersion string
+	}{
+		{
+			name:            "upgrade-v1.11.0-to-current",
+			path:            filepath.Join(*testdataDir, "hyperkit-driver-version-1.11.0"),
+			expectedVersion: "v1.11.0",
+		},
+		{
+			name:            "upgrade-v1.2.0-to-current",
+			path:            filepath.Join(*testdataDir, "hyperkit-driver-older-version"),
+			expectedVersion: version.GetVersion(),
+		},
+	}
+
+	sudoPath, err := exec.LookPath("sudo")
+	if err != nil {
+		t.Fatalf("No sudo in path: %v", err)
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			mkDir, drvPath, err := tempMinikubeDir(tc.name, tc.path)
+			if err != nil {
+				t.Fatalf("Failed to prepare tempdir. test: %s, got: %v", tc.name, err)
+			}
+			defer func() {
+				if err := os.RemoveAll(mkDir); err != nil {
+					t.Errorf("Failed to remove mkDir %q: %v", mkDir, err)
+				}
+			}()
+
+			// start "minikube start --download-only --interactive=false --driver=hyperkit --preload=false"
+			cmd := exec.Command(Target(), "start", "--download-only", "--interactive=false", "--driver=hyperkit", "--preload=false")
```

Comment:
what is reason for setting `--preload=false` here ? if not related remove that flag

Reply:
Just an optimization. Downloading the preload tarball is pretty expensive in terms of time and traffic. This flag makes the test a bit faster.

Reply:
when setting preload=false it will still download those things except they are not preloaded (slightly bigger tar file) but TestDownloadOnly runs before all the tests and downloads the preload tar file so this test will not redownload it
but if you set preload=false the extra images will be downloaded here:

 ls ~/.minikube/cache/images/k8s.gcr.io/